### PR TITLE
feat: implement condition 2 #4

### LINKED ISSUE #4: Condition 2 method

### PR DESCRIPTION
**Pull request by Storsorken**
Check that NUMPOINTS is 3 or greater is needed so that we don't get index out of bounds.

Closes #4

### CODE REVIEW

File: decide/src/main/java/org/example/Decide.java [hunk 1]
```diff
@@ -55,7 +55,9 @@ public boolean condition1() {
     }
 
     public boolean condition2() {
-        return true;
+        return settings.NUMPOINTS >= 3 && IntStream.range(0, settings.NUMPOINTS - 2).anyMatch(
+                (index) -> (settings.POINTS[index + 1].angle(settings.POINTS[index], settings.POINTS[index + 2])
+                        > Math.PI - settings.PARAMETERS.EPSILON));
```
Comment (paulinegarelli) [inline decide/src/main/java/org/example/Decide.java]: I think we're missing the other case here, it's either:
angle > PI + EPSILON
or
angle < PI - EPSILON

**Review by paulinegarelli — CHANGES_REQUESTED** [review]
Please change the comparison of the angle to include both cases, otherwise looks good!

File: decide/src/main/java/org/example/Decide.java (hunk 1 above)
Comment (Storsorken) [reply decide/src/main/java/org/example/Decide.java]: Thanks for the catch!

**Review by hampfh — APPROVED** [review]

**Review by hampfh — CHANGES_REQUESTED** [review]
sorry approved by accident

**Review by hampfh — APPROVED** [review]
👍

**Pull request merged**